torch/csrc/distributed/c10d/init.cpp

```diff
@@ -1075,7 +1075,17 @@ that adds a prefix to each key inserted to the store.
               "barrier",
               &::c10d::ProcessGroup::barrier,
               py::arg("opts") = ::c10d::BarrierOptions(),
-              py::call_guard<py::gil_scoped_release>());
+              py::call_guard<py::gil_scoped_release>())
+          .def(
+              "monitored_barrier",
+              []( const c10::intrusive_ptr<::c10d::ProcessGroup>& self, double timeoutSeconds) {
```

Comment:
We use float in RPC for timeout, I think we explicitly decided to go with float over datetime.timedelta when designing timeout for rpc framework, so thought it would be good to be consistent here. Will change it to float.

Reply:
Actually looks like we can't change it to float type since torchscript type system only supports double, and c10d APIs are now registered through torchbind:

```
caffe2/aten/src/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h:196:5: error: static_assert failed due to requirement 'guts::false_t<float>::value' "You tried to register a kernel with an unsupported input type: float. Please use double instead."
```